File: Components/Dashboard/VB/DashboardMainDemo/Data/EnergyStatictics.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Data

Namespace DashboardMainDemo
	Public NotInheritable Class EnergyStaticticsDataHelper
		Private Sub New()
		End Sub
		Public Shared Function Generate(ByVal dataSet As DataSet) As IList(Of EnergyStaticticsDataRow)
			Dim res As IList(Of EnergyStaticticsDataRow) = New List(Of EnergyStaticticsDataRow)()
			Dim data As DataRowCollection = dataSet.Tables("Countries").Rows
			For Each row As DataRow In data
				res.Add(New EnergyStaticticsDataRow With {.Country = CStr(row("Country")), .EnergyType = CStr(row("EnergyType")), .Year = CDate(row("Year")), .Import = If(row("Import") IsNot DBNull.Value, CDbl(row("Import")), 0), .Latitude = CDbl(row("Latitude")), .Longitude = CDbl(row("Longitude")), .Production = If(row("Production") IsNot DBNull.Value, CDbl(row("Production")), 0)})
			Next row
			Return res
		End Function
	End Class

	Public Class EnergyStaticticsDataRow
		Public Property Country() As String
		Public Property EnergyType() As String
		Public Property Year() As DateTime
		Public Property Import() As Double
		Public Property Latitude() As Double
		Public Property Longitude() As Double
		Public Property Production() As Double
	End Class
End Namespace

File: Components/Dashboard/CS/ASPxDashboard/App_Code/Data/EnergyStatictics.cs
using System;
using System.Collections.Generic;
using System.Data;

namespace DashboardMainDemo {
    public static class EnergyStaticticsDataHelper {
        public static IList<EnergyStaticticsDataRow> Generate(DataSet dataSet) {
            IList<EnergyStaticticsDataRow> res = new List<EnergyStaticticsDataRow>();
            DataRowCollection data = dataSet.Tables["Countries"].Rows;
            foreach(DataRow row in data) {
                res.Add(new EnergyStaticticsDataRow {
                    Country = (string)row["Country"],
                    EnergyType = (string)row["EnergyType"],
                    Year = (DateTime)row["Year"],
                    Import = row["Import"] != DBNull.Value ? (double)row["Import"] : 0,
                    Latitude = (double)row["Latitude"],
                    Longitude = (double)row["Longitude"],
                    Production = row["Production"] != DBNull.Value ? (double)row["Production"] : 0
                });
            }
            return res;
        }
    }

    public class EnergyStaticticsDataRow {
        public string Country { get; set; }
        public string EnergyType { get; set; }
        public DateTime Year { get; set; }
        public double Import { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public double Production { get; set; }
    }
}
